src/core/Akka.Remote/Transport/DotNetty/AkkaLoggingHandler.cs

```diff
@@ -8,14 +8,93 @@
 using System;
 using System.Net;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 using Akka.Util;
 using DotNetty.Buffers;
+using DotNetty.Common.Concurrency;
 using DotNetty.Transport.Channels;
 using ILoggingAdapter = Akka.Event.ILoggingAdapter;
 
 namespace Akka.Remote.Transport.DotNetty
 {
+    internal class BatchWriter : ChannelHandlerAdapter
+    {
+        private readonly int _maxPendingWrites;
+        private readonly long _maxPendingMillis;
+
+        public BatchWriter(int maxPendingWrites = 30, long maxPendingMillis = 40l)
+        {
+            _maxPendingWrites = maxPendingWrites;
+            _maxPendingMillis = maxPendingMillis;
+        }
+
+        private int _currentPendingWrites = 0;
+
+        public bool HasPendingWrites => _currentPendingWrites > 0;
+
+        public override void HandlerAdded(IChannelHandlerContext context)
+        {
+            ScheduleFlush(context);
+            base.HandlerAdded(context);
+        }
+
+        public override Task WriteAsync(IChannelHandlerContext context, object message)
+        {
+            var write = base.WriteAsync(context, message);
+            if (++_currentPendingWrites == _maxPendingWrites)
```

Comment:
A better design for doing this, potentially: since all of the messages being buffered into the `Channel` are all `IByteBuf`, we can compute the length of buffered writes for this socket thus far.

Therefore, it might be a better idea for us to change our buffering strategy to flushing when X amount of bytes are pending, rather than counting the number of messages.

That being said - if the write rate is pretty high and the message length is consistently small, we don't want to unnecessarily buffer those for too long either, so we might need a strategy that is more adaptive, based on how the channel is actually used.